easybuild/easyconfigs/a/anvio/anvio-6.1-intel-2019a-Python-3.7.2.eb

```diff
@@ -0,0 +1,171 @@
+    }),
+    ('patsy', '0.5.1', {
+        'checksums': ['f115cec4201e1465cd58b9866b0b0e7b941caafec129869057405bfe5b5e3991'],
+    }),
+    ('statsmodels', '0.10.1', {
+        'checksums': ['320659a80f916c2edf9dfbe83512d9004bb562b72eedb7d9374562038697fa10'],
+    }),
+    ('smmap2', '2.0.5', {
+        'modulename': 'smmap',
+        'checksums': ['29a9ffa0497e7f2be94ca0ed1ca1aa3cd4cf25a1f6b4f5f87f74b46ed91d609a'],
+    }),
+    ('gitdb2', '2.0.6', {
+        'modulename': 'gitdb',
+        'checksums': ['1b6df1433567a51a4a9c1a5a0de977aa351a405cc56d7d35f3388bad1f630350'],
+    }),
+    ('GitPython', '3.0.5', {
+        'modulename': 'git',
+        'checksums': ['9c2398ffc3dcb3c40b27324b316f08a4f93ad646d5a6328cafbb871aa79f5e42'],
+    }),
+    ('zipp', '0.6.0', {
+        'checksums': ['3718b1cbcd963c7d4c5511a8240812904164b7f381b647143a89d3b98f9bcd8e'],
+    }),
+    ('importlib_metadata', '0.23', {
+        'checksums': ['aa18d7378b00b40847790e7c27e11673d7fed219354109d0e7b9e5b25dc3ad26'],
+    }),
+    ('pyrsistent', '0.15.6', {
+        'checksums': ['f3b280d030afb652f79d67c5586157c5c1355c9a58dfc7940566e28d28f3df1b'],
+    }),
+    ('jsonschema', '3.2.0', {
+        'checksums': ['c8a85b28d377cc7737e46e2d9f2b4f44ee3c0e1deac6bf46ddefc7187d30797a'],
+    }),
+    ('datrie', '0.8', {
+        'checksums': ['bdd5da6ba6a97e7cd96eef2e7441c8d2ef890d04ba42694a41c7dffa3aca680c'],
+    }),
+    ('appdirs', '1.4.3', {
+        'checksums': ['9e5896d1372858f8dd3344faf4e5014d21849c756c8d5701f78f8a103b372d92'],
+    }),
+    ('ConfigArgParse', '0.15.1', {
+        'checksums': ['baaf0fd2c1c108d007f402dab5481ac5f12d77d034825bf5a27f8224757bd0ac'],
+    }),
+    ('PyYAML', '5.1.2', {
+        'modulename': 'yaml',
+        'checksums': ['01adf0b6c6f61bd11af6e10ca52b7d4057dd0be0343eb9283c878cf3af56aee4'],
+    }),
+    ('ratelimiter', '1.2.0.post0', {
+        'checksums': ['5c395dcabdbbde2e5178ef3f89b568a3066454a6ddc223b76473dac22f89b4f7'],
+    }),
+    ('wrapt', '1.11.2', {
+        'checksums': ['565a021fd19419476b9362b05eeaa094178de64f8361e44468f9e9d7843901e1'],
+    }),
+    ('snakemake', '5.8.1', {
+        'checksums': ['60c1d11d3a63397b6d91ef394639cb454d9965217747b60fafb5573a498838e4'],
+    }),
+    ('colored', '1.4.0', {
+        'checksums': ['ee8f73c40c06d9e5b829a8e284ebfaeac5ebfc7578f2eb4a0e031b40fe799a72'],
+    }),
+    ('python-Levenshtein', '0.12.0', {
+        'modulename': 'Levenshtein',
+        'checksums': ['033a11de5e3d19ea25c9302d11224e1a1898fe5abd23c61c7c360c25195e3eb1'],
+    }),
+    ('illumina-utils', '2.6', {
+        'modulename': 'IlluminaUtils',
+        'checksums': ['4ee7108d6ae67fc7d6c70bee4f775d38dfd921c10e4b020bd177838c649446ea'],
+    }),
+    ('more-itertools', '7.2.0', {
+        'checksums': ['409cd48d4db7052af495b09dec721011634af3753ae1ef92d2b32f73a745f832'],
+    }),
+
+    (name, version, {
+        'patches': ['%(name)s-%(version)s_requirements.patch'],
```

Comment:
@Darkless012 I just realized it's a lot easier to change the `requirements.txt` at "runtime" as follows:

```python
(name, version, {
    'source_urls': ['https://github.com/%(github_account)s/%(name)s/releases/download/v%(version)s/'],
    'source_tmpl': '%(name)s-%(version)s.tar.gz',
    'checksums': ['a766514d47ba012b45fef51c0ad3a810f930687c6f59531f0d2e0bd96cb05db9'],
    # replace fixed (==) versions in requirements.txt with minimal versions (>=)
    'preinstallopts': "sed -i'' 's/==/>=/g' requirements.txt && ",
}),
```

That's one less patch file to worry about... (don't forget to remove the patch file too)

Reply:
I just tried this myself, and it made the installation fail, but for good reason...

`pip check` fails with:

```
anvio 6.1 has requirement scikit-learn>=0.21.2, but you have scikit-learn 0.20.3
```

You should make sure a sufficiently recent `scikit-learn` is used, certainly newer than what's specified in `requirements.txt`.

There's two ways to fix that:

1) drop the `scikit-learn` dependency and include the latest version in `exts_list`

2) update to `intel/2019b` and Python 3.7.4, and add easyconfigs for the missing dependencies (`Python`, `SciPy-bundle`, `matplotlib` and `h5py` are already there)

I strongly prefer option 2), that's definitely the best way forward here imho...

Reply:
Updating to 2019b